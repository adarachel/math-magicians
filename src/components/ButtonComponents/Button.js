import React from 'react';
import PropTypes from 'prop-types';

function Button({ className, handleClick, name }) {
  return (
    <button className={className} type="button" onClick={handleClick}>
      {name}
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};

export default Button;
