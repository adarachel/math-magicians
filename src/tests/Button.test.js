import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../components/ButtonComponents/Button';

test('Button component renders correctly', () => {
  const { asFragment } = render(<Button name="test" handleClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Button component calls handleClick function when clicked', () => {
  const handleClickMock = jest.fn();
  const { getByText } = render(<Button name="test" handleClick={handleClickMock} />);
  const buttonElement = getByText('test');
  fireEvent.click(buttonElement);
  expect(handleClickMock).toHaveBeenCalled();
});
