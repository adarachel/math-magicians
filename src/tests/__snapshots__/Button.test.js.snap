// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    test
  </button>
</DocumentFragment>
`;
